adam/learner/object_recognizer.py

```diff
@@ -725,7 +754,142 @@ def replace_match_with_object_graph_node(
     # Example: water is a liquid
     # These may be relevant to learning verb semantics
     # (e.g. you can only drink a liquid)
-    _add_properties_linked_to_root_object_perception(
+    _add_external_properties_linked_to_root_object_perception(
+        original_graph=current_perception.copy_as_digraph(),
+        output_graph=perception_digraph,
+        matched_nodes=matched_subgraph_nodes,
+        matched_object_node=matched_object_node,
+    )
+
+    return PerceptionGraph(perception_digraph, dynamic=current_perception.dynamic)
+
+
+def replace_match_root_with_object_graph_node(
```

Comment:
I think Ryan had a (possibly better) name for this function but I can't recall what it was.